brightnessd.c

```diff
@@ -1017,6 +1030,80 @@ static uint8_t event_loop(struct Tglobalstate *pglobalstate, struct Txcb *pxcb,
     }
 }
 
+///////////////////////////////////////////////////////////////////////////////
+// parse_uint8_t
+///////////////////////////////////////////////////////////////////////////////
+/** Converts a string to an uint8_t.
+
+    @param input            the string which should be converted
+    @param output           a pointer in which the conversion result will be written
+    @return                 a non-zero value means the conversion has failed
+*/
+static int parse_uint8_t(char* input, uint8_t* output) {
+    char *end = NULL;
+    errno = 0;
+
+    long temp = strtol(input, &end, 10);
+    if (end != input && errno != ERANGE && temp >= 0 && temp <= UINT8_MAX) {
+        *output = (uint8_t)temp;
+        return 0;
+    }
+    ERROR("[parse_uint8_t] Unable to convert %s to uint8_t\n", input);
+    return 1;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// print_usage
+///////////////////////////////////////////////////////////////////////////////
+void print_usage(void) {
+    printf("Usage: brightnessd [options...]\n"
+           "\n"
+           "Available options:\n"
+           "  --cycle-brightness   PERCENTAGE               Screen brightness percentage on cycle event (X11)\n"
+           "  --timeout-brightness PERCENTAGE               Screen brightness percentage on timeout event (X11)\n"
+           );
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// parse_args
+///////////////////////////////////////////////////////////////////////////////
+/** Parse command-line argumens.
+
+    @param argc           the string which should be converted
+    @param argv           a pointer in which the conversion result will be written
+    @return               a non-zero return value indicates an error
+*/
+static int parse_args(int argc, char** argv) {
+
+    int err = 0;
+    int opt = 0;
+    static struct option long_options[] = {
+        {"cycle-brightness",   required_argument,       0,  'c' },
+        {"timeout-brightness", required_argument,       0,  't' },
+        {"help",               no_argument,             0,  'h' },
+        {0,                    0,                       0,  0   }
+    };
+
+    int long_index =0;
+    while ((opt = getopt_long(argc, argv, "c:t:h",
+                              long_options, &long_index)) != -1) {
+        switch (opt) {
+        case 'c':
+            err = parse_uint8_t(optarg, &DIM_PERCENT_INTERVAL);
+            break;
+        case 't':
+            err = parse_uint8_t(optarg, &DIM_PERCENT_TIMEOUT);
+            break;
+        case 'h':
+            print_usage();
+            exit(0);
```

Comment:
I would prefer to see `exit(EXIT_SUCCESS);` here

Reply:
Fixed